### PR TITLE
Add passkey demo

### PR DESCRIPTION
**Pull request by nicolas-camacho**
## Description
Adding demo for passkeys verification backend

## Checklist

- [x] I ran `npm test` locally and it passed without errors.
- [x] I acknowledge that all my contributions will be made under the project's [license](../blob/main/LICENSE).

## Related issues
[ACCSEC-33214](https://issues.corp.twilio.com/browse/ACCSEC-33214)

### CODE REVIEW

File: passkey-backend/README.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+`/hello-messaging` is protected and requires a valid Twilio signature as well as the following parameters:
+
+| Parameter | Description | Required |
+| :-------- | :---------- | :------- |
+
+
+## Create a new project with the template
+
+1. Install the [Twilio CLI](https://www.twilio.com/docs/twilio-cli/quickstart#install-twilio-cli)
+2. Install the [serverless toolkit](https://www.twilio.com/docs/labs/serverless-toolkit/getting-started)
```
Comment (ferazoguerrero) [inline passkey-backend/README.md]: Step 1 & 2 could be part of `Pre-requisites` section?

Comment (nicolas-camacho) [reply passkey-backend/README.md]: Yes, the readme needs to be update, on it

File: passkey-backend/functions/registration/start.js [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    });
+    return callback(null, response.data.config.creation_request);
+  } catch (error) {
+    if (error.response) {
+      console.log('Client has given an error', error);
+    } else if (error.request) {
+      console.log('Runtime error', error);
+    } else {
+      console.log(error);
+    }
```
Comment (ferazoguerrero) [inline passkey-backend/functions/registration/start.js]: error handling is repeated in different catch blocks. Is it possible to refactor it in a helper?

**Review by kmorope — APPROVED** [review]

**Review by CDavinPr1 — APPROVED** [review]

File: package.json [hunk 3]
```diff
@@ -138,6 +138,7 @@
     "voice-javascript-sdk",
     "voicemail",
     "verify-sna",
-    "flex-dialpad"
+    "flex-dialpad",
+    "passkey-backend"
```
Comment (yafuquen) [inline package.json]: Should it be `passkeys`? I recall this as a product change

File: passkey-backend/README.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
+# passkey-backend
+
+Connect appliactions with the passkey service
```
Comment (yafuquen) [inline passkey-backend/README.md]: Please use this text as the description: https://www.twilio.com/docs/verify/passkeys#twilio-verifys-support-for-passkeys

File: passkey-backend/assets/.well-know/apple-app-site-association [hunk 5]
```diff
@@ -0,0 +1,7 @@
+{
+    "webcredentials": {
+        "apps": [
+            "9EVH78F4V4.com.passkeys.twilio.sampleCode"
```
Comment (yafuquen) [inline passkey-backend/assets/.well-know/apple-app-site-association]: Do we need to send this to the repo? Should it be a placeholder and should we include the instruction in the readme to update it?

File: passkey-backend/assets/.well-know/assetlinks.json [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  {
+    "relation": [
+      "delegate_permission/common.handle_all_urls",
+      "delegate_permission/common.get_login_creds"
+    ],
+    "target": {
+      "namespace": "android_app",
+      "package_name": "com.twilio.passkeys.android",
+      "sha256_cert_fingerprints": [
+        "AF:E0:6F:5F:BF:5A:C4:E2:A0:89:22:95:B8:1C:05:4B:29:5C:80:7A:3B:69:4B:B1:0A:7B:A6:CF:1E:17:4F:A1"
```
Comment (yafuquen) [inline passkey-backend/assets/.well-know/assetlinks.json]: Do we need to send this to the repo? Should it be a placeholder and should we include the instruction in the readme to update it?

File: passkey-backend/assets/.well-know/assetlinks.json [hunk 7]
```diff
@@ -0,0 +1,64 @@
+[
+  {
+    "relation": [
+      "delegate_permission/common.handle_all_urls",
+      "delegate_permission/common.get_login_creds"
+    ],
+    "target": {
+      "namespace": "web",
+      "site": "https://6dcc8b2b3620.ngrok.app"
```
Comment (yafuquen) [inline passkey-backend/assets/.well-know/assetlinks.json]: Do we need to send this to the repo? Should it be a placeholder and should we include the instruction in the readme to update it?

File: passkey-backend/assets/.well-know/assetlinks.json [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+  {
+    "relation": [
+      "delegate_permission/common.handle_all_urls",
+      "delegate_permission/common.get_login_creds"
+    ],
+    "target": {
+      "namespace": "android_app",
+      "package_name": "com.twilio.passkeys.android",
+      "sha256_cert_fingerprints": [
+        "50:5C:D6:3D:A5:1F:19:8F:3F:B1:C2:95:0B:6B:46:B6:06:FE:C4:13:57:4B:96:7F:3D:86:A3:CF:73:FE:05:53"
```
Comment (yafuquen) [inline passkey-backend/assets/.well-know/assetlinks.json]: Same as above comment about including a placeholder

File: passkey-backend/assets/.well-know/assetlinks.json [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+  {
+    "relation": [
+      "delegate_permission/common.handle_all_urls",
+      "delegate_permission/common.get_login_creds"
+    ],
+    "target": {
+      "namespace": "android_app",
+      "package_name": "com.twilio.passkeys.android",
+      "sha256_cert_fingerprints": [
+        "E1:1B:1C:EA:3B:D8:22:EE:73:3A:26:CF:54:AF:66:0E:08:2B:CF:9A:49:7C:34:6B:D8:02:0B:6A:BA:8D:16:EF"
```
Comment (yafuquen) [inline passkey-backend/assets/.well-know/assetlinks.json]: Same as above comment about including a placeholder

File: passkey-backend/assets/.well-know/assetlinks.json [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+  {
+    "relation": [
+      "delegate_permission/common.handle_all_urls",
+      "delegate_permission/common.get_login_creds"
+    ],
+    "target": {
+      "namespace": "android_app",
+      "package_name": "com.twilio.passkey_manager",
+      "sha256_cert_fingerprints": [
+        "AF:E0:6F:5F:BF:5A:C4:E2:A0:89:22:95:B8:1C:05:4B:29:5C:80:7A:3B:69:4B:B1:0A:7B:A6:CF:1E:17:4F:A1"
```
Comment (yafuquen) [inline passkey-backend/assets/.well-know/assetlinks.json]: Same as above comment about including a placeholder

File: package.json (hunk 3 above)
Comment (yafuquen) [reply package.json]: If it is `passkeys`, please replace it in all the places where it applies, and including the name and folders of this twilio function

File: passkey-backend/functions/registration/start.js [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+      identity: event.username,
+      display_name: event.username,
+    },
+    config: {
+      relying_party: {
+        id: RELYING_PARTY,
+        name: 'PasskeySample',
+        origins: [
+          `https://${RELYING_PARTY}`,
+          'android:apk-key-hash:r-BvX79axOKgiSKVuBwFSylcgHo7aUuxCnumzx4XT6E',
```
Comment (yafuquen) [inline passkey-backend/functions/registration/start.js]: Same as above comment about including a placeholder

File: passkey-backend/README.md [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+
+In your `.env` file, set the following values:
+
+| Variable | Description | Required |
+| :------- | :---------- | :------- |
+| API_URL | Passkey API to point at | yes |
+| SERVICE_SID | Service used to call twilio services | yes |
+| RELYING_PARTY | Customer app or client | yes
+| ACCOUNT_SID | Twilio account where the service belong | yes |
+| AUTH_TOKEN | Authentication token for twilio account | yes |
```
Comment (SergioFierro) [inline passkey-backend/README.md]: Could you add this two variables to the `.env.example` file? They are missing

File: passkey-backend/README.md [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+| username | user identification name | yes
+
+
+`/registration/verification` expects the following parameters:
+
+| Parameter | Description | Required |
+| :-------- | :---------- | :------- |
+| id | A base64url encoded representation of `rawId`. | yes |
+| rawId | The globally unique identifier for this `PublicKeyCredential`. | yes |
+| type | `public-key` | yes |
```
Comment (SergioFierro) [inline passkey-backend/README.md]: I thought we decided that this is not required

File: package.json (hunk 3 above)
Comment (SergioFierro) [reply package.json]: +1, it should be `passkeys`

File: passkeys-backend/assets/.well-know/assetlinks.json [hunk 14]
```diff
@@ -0,0 +1,34 @@
...
+  {
+    "relation": [
+      "delegate_permission/common.handle_all_urls",
+      "delegate_permission/common.get_login_creds"
+    ],
+    "target": {
+      "namespace": "android_app",
+      "package_name": "com.twilio.passkey_manager",
+      "sha256_cert_fingerprints": ["{FINGERPRINT_CERTIFICATION_HASH}"]
+    }
```
Comment (SergioFierro) [inline passkeys-backend/assets/.well-know/assetlinks.json]: This relation isn't used, can you remove it please?

**Review by kmorope — APPROVED** [review]

**Review by SergioFierro — APPROVED** [review]

File: templates.json [hunk 15]
```diff
@@ -339,6 +339,11 @@
       "id": "transfers",
       "name": "Transfers",
       "description": "Transfers a call to another number"
+    },
+    {
+      "id": "passkeys-backend",
+      "name": "Backend for passkeys app",
+      "description": "Connect appliactions with the passkeys service"
```
Comment (yafuquen) [inline templates.json]: Typo `applications`

File: passkeys-backend/README.md [hunk 16]
```diff
@@ -0,0 +1,76 @@
...
+
+This project requires some environment variables to be set. A file named `.env` is used to store the values for those environment variables. To keep your tokens and secrets secure, make sure to not commit the `.env` file in git. When setting up the project with `twilio serverless:init ...` the Twilio CLI will create a `.gitignore` file that excludes `.env` from the version history.
+
+You can find a `.env.example` file to copy for creating your own `.env` file
+
+In your `.env` file, set the following values:
+
+| Variable | Description | Required |
+| :------- | :---------- | :------- |
+| API_URL | Passkey API to point at | yes |
```
Comment (yafuquen) [inline passkeys-backend/README.md]: Please change it to `Passkeys`

File: passkeys-backend/README.md [hunk 17]
```diff
@@ -0,0 +1,76 @@
...
+| Parameter | Description | Required |
+| :-------- | :---------- | :------- |
+| id | A base64url encoded representation of `rawId`. | yes |
+| rawId | The globally unique identifier for this `PublicKeyCredential`. | yes |
+| authenticatorData | A base64url encoded object given by the `AuthenticatorAttestationResponse` | yes |
+| clientDataJson | A base64url encoded object given by the `AuthenticatorAttestationResponse` | yes |
+| signature | A base64url encoded object given by the `AuthenticatorAttestationResponse` | yes |
+| userHandle | A base64url encoded object given by the `AuthenticatorAttestationResponse` | yes |
+
+## Test this project locally
```
Comment (yafuquen) [inline passkeys-backend/README.md]: Please do not include this. This tool is only for the team

File: passkeys-backend/README.md [hunk 18]
```diff
@@ -0,0 +1,76 @@
+# passkeys-backend
```
Comment (yafuquen) [inline passkeys-backend/README.md:1]: Can we add the `How to use the template`, `Create a new project with the template` and `Deploying` sections? You can find examples here: https://github.com/twilio-labs/function-templates/tree/main/verify-push-backend, https://github.com/twilio-labs/function-templates/tree/main/verify-sna, https://github.com/twilio-labs/function-templates/tree/main/verify-totp-sms

File: passkeys-backend/README.md (hunk 17 above)
Comment (nicolas-camacho) [reply passkeys-backend/README.md]: This referring to the test comment?

File: passkeys-backend/README.md (hunk 18 above)
Comment (nicolas-camacho) [reply passkeys-backend/README.md:1]: Sure

**Review by yafuquen — APPROVED** [review]

**Pull request merged**